ankaios_sdk/_components/control_interface.py

```diff
@@ -0,0 +1,365 @@
+# Copyright (c) 2024 Elektrobit Automotive GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+This script defines the ControlInterface class that handles the writing
+and reading of data to and from the Ankaios control interface.
+
+Classes
+-------
+
+- ControlInterface:
+    Handles the interaction with the Ankaios control interface.
+
+Enums
+-----
+
+- ControlInterfaceState:
+    Represents the state of the control interface.
+
+Usage
+-----
+
+- Create a Control Interface instance, connect and disconnect.
+    .. code-block:: python
+
+        ci = ControlInterface(<callbacks from Ankaios>)
+        ci.connect()
+        ...
+        ci.disconnect()
+
+- Change the state of the control interface.
+    .. code-block:: python
+
+        ci.change_state(ControlInterfaceState.CONNECTED)
+"""
+
+
+__all__ = ["ControlInterface", "ControlInterfaceState"]
+
+
+import os
+import select
+import time
+import threading
+from typing import Callable
+from enum import Enum
+from google.protobuf.internal.encoder import _VarintBytes
+from google.protobuf.internal.decoder import _DecodeVarint
+
+from .._protos import _control_api
+from .request import Request
+from .response import Response, ResponseException
+from ..exceptions import ControlInterfaceException, ConnectionClosedException
+from ..utils import DEFAULT_CONTROL_INTERFACE_PATH, get_logger, ANKAIOS_VERSION
+
+
+class ControlInterfaceState(Enum):
+    """ The state of the control interface. """
+    CONNECTED = 1
+    "(int): Connected state."
+    DISCONNECTED = 2
+    "(int): Disconnected state."
+    AGENT_DISCONNECTED = 3
+    "(int): Agent disconnected state."
+    CONNECTION_CLOSED = 4
+    "(int): Connection closed state."
+
+    def __str__(self) -> str:
+        """
+        Returns the string representation of the state.
+
+        Returns:
+            str: The state as a string.
+        """
+        return self.name
+
+
+# pylint: disable=too-many-instance-attributes
+class ControlInterface:
+    """
+    This class handles the interaction with the Ankaios control interface.
+    It provides method to send and receive data to and from the control
+    interface pipes.
+
+    Attributes:
+        logger (logging.Logger): The logger for the Ankaios class.
+        path (str): The path to the control interface.
```

Comment:
The path I want to leave it as public so that it appears in the documentation (from the docstring) and the user knows where the control interface pipes are mounted.
As for the logger, I will make it private and add the state in here as well. 
I found an issue regarding this in the Ankaios class docstring, the path was there as well, so I will also fix that.

Reply:
Changed so that the `path` variable is public and the `state` is private, but I added a property to access the state, thus not allowing you to change the state directly.